libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -235,7 +242,7 @@ - (SFSyncState*) syncDownWithTarget:(SFSyncDownTarget*)target soupName:(NSString
 - (SFSyncState*) syncDownWithTarget:(SFSyncDownTarget*)target options:(SFSyncOptions*)options soupName:(NSString*)soupName updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
     SFSyncState* sync = [SFSyncState newSyncDownWithOptions:options target:target soupName:soupName store:self.store];
     [self runSync:sync updateBlock:updateBlock];
-    return sync;
+    return [sync copy];
```

Comment:
We used to return the sync object by reference, so it could be changing before getting to javascript. For instance, with testReSync, it would sometimes already be done by the time cordova passed it to javascript causing the test to fail.